indico/modules/events/contributions/models/contributions.py

```diff
@@ -426,6 +428,11 @@ def locator(self):
     def verbose_title(self):
         return '#{} ({})'.format(self.friendly_id, self.title)
 
+    @property
+    @memoize_request
+    def paper(self):
+        return Paper(self) if self._paper_revisions else None
```

Comment:
this looks like performance hell. it requires you to always load that relationship to do *anything* PR related. I'd either use `_paper_last_revision` (you never need to load all revisions on a page listing many papers) or even unconditionally `return Paper(self)`